File: solutions/timebomb.py
# Author: Thomas Vu
# Email: thomas.vu@ucalgary.ca
# Feel free to send any questions about this problem to the email above
# or ask in the CPC discord. (discord.gg/MEXwfze)
# ---------------------------------------------------------------------

asciiNums = ["**** ** ** ****",
             "  *  *  *  *  *",
             "***  *****  ***",
             "***  ****  ****",
             "* ** ****  *  *",
             "****  ***  ****",
             "****  **** ****",
             "***  *  *  *  *",
             "**** ***** ****",
             "**** ****  ****"]

line1 = input()
line2 = input()
line3 = input()
line4 = input()
line5 = input()

numDigits = (len(line1) + 1) // 4
numbers = []
for _ in range(numDigits):
    numbers.append("")

for i in range(numDigits):
    numbers[i] += line1[i*4:i*4+3]
    numbers[i] += line2[i*4:i*4+3]
    numbers[i] += line3[i*4:i*4+3]
    numbers[i] += line4[i*4:i*4+3]
    numbers[i] += line5[i*4:i*4+3]

theNumber = ""
boom = False
for n in numbers:
    if n in asciiNums:
        theNumber += str(asciiNums.index(n))
    else:
        boom = True

if boom == False and int(theNumber) % 6 == 0:
    print("BEER!!")
else:
    print("BOOM!!")

File: solutions/Timebomb.java
/*
 Author: Martin Tran
 Email: martin.tran@ucalgary.ca

 Feel free to send any questions about this problem to the email
 above or ask in the CPC discord. (discord.gg/MEXwfze)
 */
import java.util.Scanner;

public class Timebomb {

    public static void main(String args[]) {
	// ASCII 0-9. Each line is one number.
	String[] numbers = {"**** ** ** ****",
			    "  *  *  *  *  *",
			    "***  *****  ***",
			    "***  ****  ****",
			    "* ** ****  *  *",
			    "****  ***  ****",
			    "****  **** ****",
			    "***  *  *  *  *",
			    "**** ***** ****",
			    "**** ****  ****"};

	String[] bombcode = {"", "", "", "", "", "", "", ""};

	Scanner sc = new Scanner(System.in);
	String line;
	for (int i = 0; i < 5; i++) {
	    line = sc.nextLine() + " ";
	    for (int j = 0; j < line.length()/4; j++)
		bombcode[j] += line.substring(4*j, (4*(j+1))-1);
	}

	String translated_code = "";
	for (String s : bombcode) {
	    if (s.equals(numbers[0]))
		translated_code += "0";
	    else if (s.equals(numbers[1]))
		translated_code += "1";
	    else if (s.equals(numbers[2]))
		translated_code += "2";
	    else if (s.equals(numbers[3]))
		translated_code += "3";
	    else if (s.equals(numbers[4]))
		translated_code += "4";
	    else if (s.equals(numbers[5]))
		translated_code += "5";
	    else if (s.equals(numbers[6]))
		translated_code += "6";
	    else if (s.equals(numbers[7]))
		translated_code += "7";
	    else if (s.equals(numbers[8]))
		translated_code += "8";
	    else if (s.equals(numbers[9]))
		translated_code += "9";
	    else if (s.equals(""))
		continue;
	    else {
		System.out.println("BOOM!!");
		return;
	    }
	}
	
	if ((Integer.parseInt(translated_code) % 6) == 0)
	    System.out.println("BEER!!");
	else
	    System.out.println("BOOM!!");
    }
}
